### ISSUE TITLE
How to support multiple routes in index.php file?

### ISSUE THREAD

**Issue by semaurer** [devloco/create-react-wptheme#10]
Hi,

This is a question not an issue.

I am using react router and have a route called /example1
If I refresh the page while on /example1 locally it works, because the wp-dev server is running, and it serves up the react app.  

But when the build is deployed, there's nothing to catch the request when you refresh the page on routes other than "/", so you get a blank page.  How can I add additional route support to the php file beyond the root "/"? Or would it be in the index.html file?

Thank you.

**Comment by devloco** [comment]
Hi Stephen,

**TL;DR:** there was a bug in the "npm run build" command a couple of weeks ago. Check out the [Updating](https://github.com/devloco/create-react-wptheme#updating) instructions in the README and do that first. Then try modifying the user.prod.json file per the [Build Configuration](https://github.com/devloco/create-react-wptheme#build-configuration) instructions. Hopefully `npm run build` will just work correctly for you then.

Here was the issue ticket for that bug, if you're interested: [user.prod.json homepage property not used on wpbuild](https://github.com/devloco/create-react-wptheme/issues/9)

If that doesn't just work for you, then you're going to have to read **The Long Answer** ;)

First off, there is no "wp-dev server" serving your pages. That's the whole point of this project, it is your WordPress server serving your theme. There is a little web-socket server running, but all it does it tell your browser that you made changes and then your browser refreshes when that happens.

**So the fact that it works in your dev environment leads me to hope that you're just using the old version with the build bug. Sorry about that**!

If updating and **double-checking** your `user.prod.json` file doesn't work, then I'm going to be a little stuck trying to help... I'm not much of a React-Router expert. But you might look at this tutorial, the author does use React Router in his theme: [How To Build A Skin For Your Web App With React And WordPress](https://www.smashingmagazine.com/2018/03/react-wordpress-web-app/)

Finally, there might be some kind of **hack** you can come up with using either Webpack's "%PUBLIC_URL%" feature (see references to that in the index.php file in your public folder) or getting your theme's URL using PHP, like I do here in this file: [post_installer.php](https://github.com/devloco/create-react-app/blob/a1446acc585a5426900a8210a13ffceb7df5696a/packages/react-scripts/template/public/post_installer.php#L2). **I do think that using either method is indeed a hack, so be careful.**

Let me know how it goes!

Thanks,

-joe

**Comment by semaurer** [comment]
Thanks for your response and help!

Looks like the culprit was actually just the wordpress site not being live.  So when I refreshed on 
example.com/~wpuser/examplepage.  It would break just due to the url not matching, because my react router replacement was assuming that there would be no /~wpuser in the url.  Once live without the ~wpuser it works fine when refreshing on /example page.

Thank you!
